### PR TITLE
feat(ecma-analyzer): support super() call in class ctor

### PR DESCRIPTION
**Pull request by kwonoj**
### Description

Attempt to close WEB-1089.

PR tries to support import resolustion inside of `super` call, if given args are referencing esm imports.

Since super() is sort of expr_call, I originally tried to reuse JsValue::Call as-is but not successfully. Still that might be a simple mistake and new addition to JsValue::Super may not be required.

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]

File: crates/turbopack-ecmascript/src/analyzer/mod.rs [hunk 1]
```diff
@@ -421,6 +421,9 @@ pub enum JsValue {
     /// A function call without a this context.
     /// `(total_node_count, callee, args)`
     Call(usize, Box<JsValue>, Vec<JsValue>),
+    /// A super call to the parent constructor.
+    /// `(total_node_count, args)`
+    Super(usize, Vec<JsValue>),
```
Comment (sokra) [inline crates/turbopack-ecmascript/src/analyzer/mod.rs:426]:
```suggestion
    SuperCall(usize, Vec<JsValue>),
```

File: crates/turbopack-ecmascript/src/analyzer/graph.rs [hunk 2]
```diff
@@ -898,7 +913,13 @@ impl Analyzer<'_> {
                     });
                 }
             }
-            _ => {}
+            Callee::Super(_) => self.add_effect(Effect::Call {
+                func: JsValue::FreeVar(js_word!("super")),
```
Comment (sokra) [inline crates/turbopack-ecmascript/src/analyzer/graph.rs:917]:
```suggestion
                func: JsValue::Super,
``

super is not a free var. It should have a special token, or stay Unknown

File: crates/turbopack-ecmascript/src/analyzer/mod.rs [hunk 3]
```diff
@@ -811,6 +823,10 @@ impl JsValue {
         Self::Call(1 + f.total_nodes() + total_nodes(&args), f, args)
     }
 
+    pub fn callee_super(args: Vec<JsValue>) -> Self {
```
Comment (sokra) [inline crates/turbopack-ecmascript/src/analyzer/mod.rs:826]:
```suggestion
    pub fn super_call(args: Vec<JsValue>) -> Self {
```

**Review by sokra — CHANGES_REQUESTED** [review]
add a test case

**Review by jridgewell — APPROVED** [review]

**Pull request merged**